### PR TITLE
Conversation page -> DM

### PR DESCRIPTION
**Pull request by abigailshilts**
-Creates conversation page that displays a list of all current conversations
-Will segue to the individual conversation from conversation page
-Two users can communicate in real time between the apps when both are logged in

### CODE REVIEW

**Review by nsazonov — COMMENTED** [review]
Please add .DS_Store to .gitignore file. This might be a good example https://stackoverflow.com/questions/49478/git-ignore-file-for-xcode-projects

File: PickMeUp/PickMeUp/PMConversation.h [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
 //  Created by Abigail Shilts on 7/14/22.
 //
-
-#import <Parse/Parse.h>
 #import <Foundation/Foundation.h>
+#import <Parse/Parse.h>
 NS_ASSUME_NONNULL_BEGIN
 
-@interface Conversation : PFObject<PFSubclassing>
+@interface PMConversation : PFObject<PFSubclassing>
```
Comment (nsazonov) [inline PickMeUp/PickMeUp/PMConversation.h:11]: Looks good, glad you're adding prefix for Objective C names

File: PickMeUp/PickMeUp/PMConversation.h [hunk 2]
```diff
@@ -4,15 +4,15 @@
...
 @property (nonatomic, strong) PFUser *user1;
 @property (nonatomic, strong) PFUser *user2;
-- (void)postConvo:(PFUser *)user otherUser:(PFUser *)otherUser withCompletion: (PFBooleanResultBlock  _Nullable)completion;
+- (void)postConvo:(PFUser *)user otherUser:(PFUser *)otherUser
+   withCompletion: (PFBooleanResultBlock  _Nullable)completion;
```
Comment (nsazonov) [inline PickMeUp/PickMeUp/PMConversation.h:15]: `withCompletion:(PFBooleanResultBlock  _Nullable)completion;`

spacing inconsistency

File: PickMeUp/PickMeUp/PMConversation.m [hunk 3]
```diff
@@ -5,24 +5,25 @@
...
+#import "StringsList.h"
 
-@implementation Conversation
+@implementation PMConversation
 @dynamic user1;
 @dynamic user2;
 
 + (nonnull NSString *)parseClassName {
-    return @"Conversation";
+    return kConversationClassName;
```
Comment (nsazonov) [inline PickMeUp/PickMeUp/PMConversation.m:17]: 👍

File: PickMeUp/PickMeUp/PMConversationViewController.m [hunk 4]
```diff
@@ -5,33 +5,39 @@
...
 - (void)viewDidLoad {
     [super viewDidLoad];
-    self.recieverName.text = self.reciever.username;
+    self.recieverName.text = self.receiver.username;
     self.tableView.dataSource = self;
     self.tableView.delegate = self;
     self.tableView.rowHeight = UITableViewAutomaticDimension;
     self.messageToSend.delegate = self;
-    [self getQuery];
+    [NSTimer scheduledTimerWithTimeInterval:5 target:self selector:@selector(onTimer) userInfo:nil repeats:true];
```
Comment (nsazonov) [inline PickMeUp/PickMeUp/PMConversationViewController.m:35]: Are you polling the server while user is on the conversation screen? Move 5 to a constant like kConversationPollIntervalInSeconds

Comment (nsazonov) [reply PickMeUp/PickMeUp/PMConversationViewController.m:35]: I'd recommend to same NSTimer instance and stop the timer when it's not needed. But maybe you always want it running, not sure.

File: PickMeUp/PickMeUp/PMConversationViewController.m [hunk 5]
```diff
@@ -74,28 +80,30 @@ - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger
 
 -(void)saveDM {
     // posts DM to database and refreshes the table
-    DirectMessage *newDM = [DirectMessage new];
+    self.mybtn.enabled = NO;
```
Comment (nsazonov) [inline PickMeUp/PickMeUp/PMConversationViewController.m:83]: let's rename the button to something more meaningful

File: PickMeUp/PickMeUp/PMDMCell.m [hunk 6]
```diff
@@ -11,13 +11,10 @@ @implementation PMDMCell
 
 - (void)awakeFromNib {
```
Comment (nsazonov) [inline PickMeUp/PickMeUp/PMDMCell.m:12]: if the method is empty just delete it

File: PickMeUp/PickMeUp/PMDMCell.m [hunk 7]
```diff
@@ -11,13 +11,10 @@ @implementation PMDMCell
...
     [super awakeFromNib];
-    // Initialization code
 }
 
 - (void)setSelected:(BOOL)selected animated:(BOOL)animated {
```
Comment (nsazonov) [inline PickMeUp/PickMeUp/PMDMCell.m:16]: same here

File: PickMeUp/PickMeUp/PMDisplayConversationsViewController.m [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+static const NSString *const kConvoCell = @"convoCell";
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    self.tableView.rowHeight = 80; // needs to be changed later to make row height responsive to content
+    self.currentUser = [PFUser currentUser];
+    [self getQuery];
```
Comment (nsazonov) [inline PickMeUp/PickMeUp/PMDisplayConversationsViewController.m:29]: nit: runGetQuery

**Review by nsazonov — APPROVED** [review]

File: PickMeUp/PickMeUp/PMConversation.m [hunk 9]
```diff
@@ -5,24 +5,25 @@
 //  Created by Abigail Shilts on 7/14/22.
 //
 
-#import "Conversation.h"
+#import "PMConversation.h"
 #import "Parse/Parse.h"
```
Comment (czamorag) [inline PickMeUp/PickMeUp/PMConversation.m:9]: nit: Imports should be in alphabetical order

File: PickMeUp/PickMeUp/PMConversationViewController.h [hunk 10]
```diff
@@ -4,16 +4,15 @@
 //
 //  Created by Abigail Shilts on 7/14/22.
 //
-
+#import "PMConversation.h"
```
Comment (czamorag) [inline PickMeUp/PickMeUp/PMConversationViewController.h:7]: Same, alphabetical order and preferably put the “<>“ imports together

File: PickMeUp/PickMeUp/PMDetailsViewController.m [hunk 11]
```diff
@@ -7,8 +7,8 @@
 
 #import "PMDetailsViewController.h"
 #import "UIImageView+AFNetworking.h"
-#import "ConversationViewController.h"
-#import "Conversation.h"
+#import "PMConversationViewController.h"
```
Comment (czamorag) [inline PickMeUp/PickMeUp/PMDetailsViewController.m:10]: Same, order

File: PickMeUp/PickMeUp/StringsList.h [hunk 12]
```diff
@@ -8,32 +8,32 @@
...
-static const NSString *const plist = @"plist";
-static const NSString *const postCell = @"postCell";
-static const NSString *const postedSuccess = @"posted photo!";
-static const NSString *const serverLink = @"https://parseapi.back4app.com";
-static const NSString *const strInput = @"%@";
-static const NSString *const upAny = @"Any";
-static const NSString *const userRegSuccess = @"User registered successfully";
-static const NSString *const user1 = @"user1";
-static const NSString *const user2 = @"user2";
+static const NSString *const kAppIDString = @"appId";
```
Comment (czamorag) [inline PickMeUp/PickMeUp/StringsList.h:11]: I think you should create 2 different string files: One just for strings that can be localized and the second one for the rest of constants that will be used along the app

**Pull request merged**